qa/vagrant/build.gradle

```diff
@@ -92,6 +71,24 @@ repositories {
 }
 
 dependencies {
+  String maybeTestsSeed = System.getProperty("tests.seed", null);
+  if (maybeTestsSeed != null) {
+    seed = new BigInteger(maybeTestsSeed, 16).longValue()
+    formattedSeed = maybeTestsSeed
+  } else {
+    seed = new Random().nextLong()
+    formattedSeed = String.format("%016X", seed)
+  }
+
+  String maybeUpdradeFromVersions = System.getProperty("tests.packaging.upgrade.from.versions", null)
+  if (maybeUpdradeFromVersions != null) {
+    upgradeFromVersions = maybeUpdradeFromVersions.split(",")
+  } else {
+    upgradeFromVersions = new File(project.projectDir, 'src/main/resources/versions')
```

Comment:
I'm fairly sure the dependencies closure is executed immediately and we try not to do IO there. But @rjernst will know what is the most right thing to do here. It could be as simple as making a method for this and calling it from within the string on line 104 and 105 with `${ -> upgradeFromVersion() }`